### ISSUE TITLE
Breeze: Build documentation with Breeze

### ISSUE THREAD

**Issue by potiuk** [apache/airflow#20739]
The "build-documentation" action should be rewritten in python in the new Breeze2 command.

It should allow for the same parameters that are already used by the https://github.com/apache/airflow/blob/main/docs/build_docs.py script: `--package-filter` for example.

It shoudl basically run the ./build_docs.py using CI image.

Also we should add `airflow-start-doc-server` as a separate entrypoint (similar to airflow-freespace).

**Comment by mik-laj** [comment]
For the sake of clarity, I would like to add that we want to keep the './docs/build_docs.py' file because it is used for building documentation locally and we should not make any redundant changes to it.

**Comment by potiuk** [comment]
Absolutely. ~./Breeze docs build will be particularly useful for people who either do not have local venv specifically prepared for building docs.

**Comment by MatrixManAtYrService** [comment]
Not sure if this is in-scope here, but my impression is that `./docs/start_doc_server.sh` doesn't work if `python` doesn't refer to `python3`.  Might make a decent "while-you're-at-it" kind of change.

**Comment by potiuk** [comment]
> Not sure if this is in-scope here, but my impression is that `./docs/start_doc_server.sh` doesn't work if `python` doesn't refer to `python3`. Might make a decent "while-you're-at-it" kind of change.

I think we might actually move the "start-doc-server" to be part of the new Breeze. The new Breeze2 nicely works with `pipx` in the way that it will install entrypoints configured in https://github.com/apache/airflow/blob/3b5adaff9a30bee3ee12b32c44f38c3f5148df24/dev/breeze/setup.cfg#L67 as scripts available on the path (independently if you are on windows/osx/linux).

I think we should convert all our "standalone" scripts to be installed with `pipx --install ./dev/breeze`  - which will become a nice entrypoint for buiding and using the Breeze/CI image but also for various tools we are using (for example freespace which we us on CI in the same way).

This is actually very nice thing to have.

**Comment by Bowrna** [comment]
> [WIP] Check free space python #20701



> Absolutely. ~./Breeze docs build will be particularly useful for people who either do not have local venv specifically prepared for building docs.

@potiuk Does this means that Breeze should start docker command to create container alone and the execution of script (build_docs.py) is taken care from there? ( so that it works independent of venv)

**Comment by Bowrna** [comment]
```
./breeze build-docs -- --package-filter apache-airflow-providers-amazon
```
Why the args are passed after `--` though there is `--` prefixing the option? @potiuk

**Comment by potiuk** [comment]
> Why the args are passed after `--` though there is `--` prefixing the option? @potiuk

The `--` is a way to indicate parameters passed directly to underlying tool - at that time we did not want to define all the parameters as "build-docs" command parameters so we passed them through directly to the build-docs.sh script.

This is the typical/standard way of passing parametrs as pass -through to underlyig command. The parameters before `--` are interpreted by the `./breeze build-docs` command and the parameter after `--` are pased to the ./build-docs.sh script executed under-the-hood (via "${@}").

In the original breeze I did not want to explode even more the number of options - that's why it was done this way. But in the new Breeze. We should define those parameters as "./breeze build-docs` command parameters so that we can get them auto-completeable.

So in the new Breeze2 it should be:

`./Breeze2 build-docs --package-filter ....` (See the ./build-docs.sh script and the parameters it takes).


I actually think about even more consolidation of those commands but I will formalize and propose it next week.
